iceoryx_posh/test/integrationtests/test_roudi_findservice.cpp

```diff
@@ -41,10 +41,11 @@ class RoudiFindService_test : public RouDi_GTest
 
 TEST_F(RoudiFindService_test, OfferSingleMethodServiceSingleInstance)
 {
-    auto isServiceOffered = senderRuntime->offerService({"service1", "instance1"});
+    auto isServiceOffered = senderRuntime->offerService({"service1", "instance1", "event1"});
     this->InterOpWait();
 
-    auto instanceContainer = receiverRuntime->findService({"service1", "instance1"});
+    auto instanceContainer = receiverRuntime->findService({iox::cxx::in_place_type<IdString_t>(), "service1"},
+                                                          {iox::cxx::in_place_type<IdString_t>(), "instance1"});
```

Comment:
If there are no objections regarding this interface, I will add a generic c'tor to `cxx::variant` in another PR to simplify the usage to the following:
```
enum class SearchMode
{
    WILDCARD
};

void findService(const std::variant<SearchMode, cxx::string> service, const std::variant<SearchMode, cxx::string> instance)
{
   //.. snip ..
}

int main()
{
    findService("Blubb", SearchMode::WILDCARD);
}
```

This is [supported](https://godbolt.org/z/q3dvnvEMG) by the `std::variant`, [c'tor 4](https://en.cppreference.com/w/cpp/utility/variant/variant) but currently not by our `cxx::variant`.

Reply:
First version [worked](https://github.com/ApexAI/iceoryx/commit/705525d10e28d29b0d4a731c164c44e534e97a15).

Reply:
In case we have
```cpp
void foo(cxx::variant<int, bool, int> a);
```
What would happens with
```
foo(42);
```
I guess the same as with
```cpp
foo({iox::cxx::in_place_type<int>(), 42})
```